project.md

```diff
@@ -11,41 +11,61 @@ _What need is met by your idea? Who are you helping? What is that person's
+
 ### Why you?
 
 _What excites you about this idea? How did you come up with it?_
 
+I've loved choose-your-own-adventure books and text-based adventure games since I was a kid. As a project, it would give me a better appreciation for the medium while working on the skills we've covered so far in this course. I thought of it while considering the kinds of project that wouldn't be too implementation-heavy while giving lots of leeway on the language design. I was thinking about everything that can be accomplished within a terminal window, and I was reminded of these kinds of games.  
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Developing simple text-based adventure games!
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+The overall philosophy with my design is to make it as lightweight as possible, so writers can focus on writing and not on juggling syntax. My plan is to take influence from Markdown and XML to make something that is not terribly intrusive yet powerful enough for the features I wish to implement (room, interaction, and inventory system on the backend along with related GO, LOOK, TALK, TAKE, USE, and INVENTORY commands for the player to use in the terminal). Tags and nesting from XML will allow for rooms to be established with different objects, people, and items within them, while certain symbols and parentheses around tags (like in Markdown) can define relationships between rooms and between objects. This is the right way to work in this domain because it implies the necessary relationships between all the things in the game while removing all the logic that would otherwise be necessary to implement yourself.
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When the program runs, the game as programmed will start on the command line. This will involve printing things to the terminal and waiting for user input, where the process repeats. Some common errors will likely be improperly closed tags, bad nesting, and relationship references to tags that do not exist. My DSL will need good error-catching to be able to diagnose these problems and return clear and relevant error messages to the user in the terminal. I'll need to describe the problem in clear terms and give a relevant suggestion for how to fix it! 
```

Comment:
You correctly recognize that debugging can be really difficult for novice coders, but it's also important to remember that it's difficult to implement custom error handling in an external DSL, so i'm interested to see what strategies you use!